src/playback.jl

```diff
@@ -0,0 +1,45 @@
+
+"""
+    RandomPolicy{RNG<:AbstractRNG, P<:Union{POMDP,MDP}, U<:Updater}
+a generic policy that uses the actions function to create a list of actions and then randomly samples an action from it.
+
+Constructor:
+
+    `PlaybackPolicy(actions::Vector{A}, backup_policy::Policy; logpdfs::Vector{Float64} = [])`
+
+# Fields
+- `actions::Vector{A}` a vector of actions to play back
+- `backup_policy::Policy` the policy to use when all prescribed actions have been taken but the episode continues
+- `logpdfs::Vector{Float64}` the log probability (density) of actions
+- `i::Int64` the current action index
+"""
+mutable struct PlaybackPolicy{A} <: Policy
+    actions::Vector{A}
+    backup_policy::Policy
```

Comment:
the policy type should be part of the parameters to make it a concrete type: 
```julia
struct PlaybackPolicy{A, P<:Policy} <: Policy 
    actions::Vector{A}
    backup_policy::P
    logpdfs::Vector{Float64} 
    i::Int64
end
```

Reply:
Even better would be to use `AbstractArray` instead of `Vector` for `actions` and `logpdfs`

Reply:
Agreed, thanks for pointing that out!